R/teal_slice.R

```diff
@@ -481,50 +520,58 @@ c.teal_slices <- function(...) {
   )
 }
 
+#' @rdname teal_slice
+#' @keywords internal
+#'
+slices_to_list <- function(tss) {
+  slices_list <- lapply(tss, as.list)
+  attrs <- attributes(unclass(tss))
+  tss_list <- list(slices = slices_list, attributes = attrs)
+  Filter(Negate(is.null), tss_list) # drop attributes if empty
+}
 
-# format method for teal_slices
+#' @param show_all `logical(1)` should parameters set to NULL be returned
+#' @param nchars `integer(1)` or `NULL` indicating the number of character of the output width
 #' @export
 #' @rdname teal_slice
 #' @keywords internal
 #'
-format.teal_slices <- function(x, show_all = FALSE, ...) {
-  res <- character(0)
-  for (i in seq_along(x)) {
-    ind <- names(x)[i]
-    if (is.null(ind)) ind <- sprintf("[[%d]]", i)
-    res <- append(res, ind)
-    res <- append(res, format(x[[i]], show_all = show_all, ...))
-  }
-
-  includes <- attr(x, "include_varnames")
-  if (length(includes) > 0L) {
-    res <- append(res, "\nfilterable variables:")
-    for (i in seq_along(includes)) {
-      res <- append(res, sprintf(" $ %s: %s", names(includes)[i], toString(includes[[i]])))
+format.teal_slices <- function(x, show_all = FALSE, nchars = 15, ...) {
+  checkmate::assert_flag(show_all)
+  checkmate::assert_integerish(nchars, null.ok = TRUE)
+
+  slices_format_name <- function(name, name_width) {
```

Comment:
I wonder if we need name formatting in `teal_slices` anyway. 
All `teal_slice` need not be justified to the same position. We can call `format` on `teal_slice` elements and have `teal_slice` justified. We can live with attributes not being adjusted in my opinion but we can explicitly call `format.teal_slice` on the attributes. Lines that only contain braces, brackets, and commas are not modified anyway.

What do you think?

Reply:
yeah @chlebowa but we decided to separate `format.teal_slices` so it does not use `format.teal_slice` becuase when we wanted to use `format.teal_slice` in `format.teal_slices` we ended up cleaning JSON character manually as we no longer were working on full JSON objects but on trimmed characters. separate slices_format_name is the consequence we don't use `format.teal_slice` in `format.teal_slices`

Reply:
Riiiight, I forgot about that...

Reply:
fixed by chlebowa in next commits